src/packagedcode/phpcomposer.py

```diff
@@ -0,0 +1,403 @@
+#
+# Copyright (c) 2015 nexB Inc. and others. All rights reserved.
+# http://nexb.com and https://github.com/nexB/scancode-toolkit/
+# The ScanCode software is licensed under the Apache License version 2.0.
+# Data generated with ScanCode require an acknowledgment.
+# ScanCode is a trademark of nexB Inc.
+#
+# You may not use this software except in compliance with the License.
+# You may obtain a copy of the License at: http://apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+#
+# When you publish or redistribute any data created with ScanCode or any ScanCode
+# derivative work, you must accompany this data with the following acknowledgment:
+#
+#  Generated with ScanCode and provided on an "AS IS" BASIS, WITHOUT WARRANTIES
+#  OR CONDITIONS OF ANY KIND, either express or implied. No content created from
+#  ScanCode should be considered or used as legal advice. Consult an Attorney
+#  for any legal advice.
+#  ScanCode is a free software code scanning tool from nexB Inc. and others.
+#  Visit https://github.com/nexB/scancode-toolkit/ for support and download.
+
+from __future__ import absolute_import
+from __future__ import print_function
+
+import codecs
+from collections import OrderedDict
+from functools import partial
+import json
+import logging
+import re
+
+from commoncode import filetype
+from commoncode import fileutils
+
+from packagedcode import models
+
+"""
+Handle PHP composer packages
+per https://getcomposer.org/
+"""
+
+"""
+To check
+https://github.com/pombredanne/normalize-package-data
+"""
+
+
+logger = logging.getLogger(__name__)
+# import sys
+# logging.basicConfig(level=logging.DEBUG, stream=sys.stdout)
+# logger.setLevel(logging.DEBUG)
+
+
+class PHPComposerPackage(models.Package):
+    metafiles = ('package.json')
+    filetypes = ('.json',)
+    mimetypes = ('application/x-tar',)
+    repo_types = (models.repo_phpcomposer,)
+
+    type = models.StringType(default='phpcomposer')
+    primary_language = models.StringType(default='JavaScript')
+
+    @classmethod
+    def recognize(cls, location):
+        return parse(location)
+
+
+def is_phpcomposer_json(location):
+    return (filetype.is_file(location)
+            and fileutils.file_name(location).lower() == 'composer.json')
+
+
+def parse(location):
+    """
+    Return a Package object from a composer.json file or None.
+    """
+    if not is_phpcomposer_json(location):
+        return
+    # mapping of top level composer.json items to the Package object field name
+    plain_fields = OrderedDict([
+        ('name', 'name'),
+        ('description', 'summary'),
```

Comment:
What about the version? Why do you handle it differently  below than these other attributes?